ERA/ERi.nls

```diff
@@ -1,43 +1,96 @@
-  let i (list 0 1 2 3)
-  let ERiC-list table:to-list ERiC ;; [["currency" ERiC][...][...][...]]
-  ;; first foreach loops over sets of exchange rates (so the set of reference currencies)
-  foreach i [ index ->
-    let base-exchange-rates-table item index BASE-YEAR-EXCHANGE-RATES
-    let base-exchange-rates-list table:to-list base-exchange-rates-table ;; [["target currency" exchange rate][...][...][...]]
-    let ERiC-ref last (item index ERiC-list) ;; one number, ERiC of reference currency
-    let new-exchange-rates-list (list)
-  ;; second foreach loops over individual exchange rates within a set
-    let index-counter 0
-    foreach base-exchange-rates-list [ entry ->
-      let currency-name first entry
-      let base-rate last entry ;; 1 reference currency = (base-rate) target currency
-      let ERiC-target last (item index-counter ERiC-list)
-      let multiplier (ERiC-ref / ERiC-target)
-      let new-rate precision (base-rate * multiplier) 4
-      table:put item index CURRENT-EXCHANGE-RATES currency-name new-rate
-      set index-counter (index-counter + 1)
+
+  let curr-currency 0
+  while [curr-currency < length CURRENCY-VALUES] [
+    set CURRENCY-VALUES replace-item curr-currency CURRENCY-VALUES [ERiC] of one-of ops-nodes with [who = curr-currency] ;;theres only one, ignore the one-of used to grab the only ops-node
+    set curr-currency curr-currency + 1
+  ]
+  let base-exchange-rates (list)
+  let currency-row 0
+  let exch-rate table:make
+  while [currency-row < num-jurisdictions + num-decentralized-currencies] [
+    let currency-column 0
+    while [currency-column < num-jurisdictions + num-decentralized-currencies] [
+      let exchange item currency-column CURRENCY-VALUES / item currency-row CURRENCY-VALUES
+      table:put exch-rate currency-column exchange
+      set currency-column currency-column + 1
+    ]
+    set currency-row currency-row + 1
+    set base-exchange-rates lput (copy-table exch-rate) base-exchange-rates
+  ]
+  set BASE-YEAR-EXCHANGE-RATES base-exchange-rates
+end
+
+to ERi.recalculate-ERiE ;called by observer
+  let new-ERiE table:make
+  let soil-health-table table:make
+  let eco 0
+  let curr-soil-health 0
+  let curr-eco "A"
+  while [eco < num-ecoregions] [
+    set curr-eco (index-to-ecoregion eco)
+    ifelse count patches with [ecoregion = curr-eco] = 0 [
+      set curr-soil-health 0
+    ]  [
+      set curr-soil-health min (list 1 ((mean [soil-health] of patches with [ecoregion = curr-eco] - max-soil-health / 2) / max-soil-health / 2)) ;;normalizes worst case to -1 and best case to 1
     ]
+    table:put soil-health-table curr-eco curr-soil-health
+    table:put new-ERiE curr-eco (Bezier (curr-soil-health) 0 1)
+    set eco eco + 1
+  ]
+  set PREVIOUS-SOIL-HEALTH soil-health-table
+  set ERiE new-ERiE
+end
+
+to ERi.recalculate-ERiC ;called by ops-nodes
+  let eco 0
+  let eric-temp 0
+  while [eco < num-ecoregions] [
```

Comment:
should probably be a foreach ecoregions (maybe you should store ecoregions in a global to make it easy to foreach across them).   Also, would it be easier to make a  new turtle breed with one turtle for each ecoregion to store info for ecoregions? It would be similar to an ops-node for a  jurisdiction but it's only purpose would be to store info about an eco-region. Let me know what you think.

Reply:
Other than ERiC, does the model contain any other ecoregion specific data?